### PR TITLE
WiimoteEmu: Allow shake frequency and intensity to be configured.

### PR DESCRIPTION
**Pull request by jordan-woyak**
Screenshot: https://i.imgur.com/ASLrUrL.png

Frequency setting controls the approximate shakes per second.
Intensity setting controls the total travel in centimeters.

Partially pressing inputs will reduce intensity.

A fancy graph shows the effects on the emulated wiimote's position.

Donkey Kong Country Returns expects wiimote and nunchuk to be shaken in opposing directions which can now be achieved with a negative intensity value on either Wiimote or Nunchuk.
Increasing the frequency to about 12 kinda seems to force it to work as well which is why it used to work.
Fixes: https://bugs.dolphin-emu.org/issues/11535

Shake inputs now affect IR data.

Defaults:
Intensity: 10 cm (-10 for Nunchuk)
Frequency: 6 hz

This works with the problem games I am aware of:
Batman: TBATB
Jett Rocket
Donkey Kong Country Returns

### CODE REVIEW

**Comment by Miksel12** [comment]
Works great in DKCR.
Only problem I have is that it looks a bit weird in SMG with the IR now affected by shake, could this be made toggleable?

File: Source/Core/Core/HW/WiimoteEmu/Dynamics.cpp [hunk 1]
```diff
@@ -60,81 +53,30 @@ double CalculateStopDistance(double velocity, double max_accel)
...
-                                 ControllerEmu::Buttons* const buttons_group,
-                                 const DynamicConfiguration& config, u8* const shake_step)
-{
-  // shake is a bitfield of X,Y,Z shake button states
-  static const unsigned int btns[] = {0x01, 0x02, 0x04};
-  unsigned int shake = 0;
-  buttons_group->GetState(&shake, btns);
-
-  Common::Vec3 accel;
+  const auto peak_accel_time = 1 / shake_group->GetFrequency() / 2 / 2;
```
Comment (BhaaLseN) [inline Source/Core/Core/HW/WiimoteEmu/Dynamics.cpp]: This `/ 2 / 2` looks odd, any particular reason why you wrote it like that?

Comment (jordan-woyak) [reply Source/Core/Core/HW/WiimoteEmu/Dynamics.cpp]: I guess it's just the way my head was struggling at the math. I'll change it to `/ 4`.

**Comment by jordan-woyak** [comment]
> Works great in DKCR.
> Only problem I have is that it looks a bit weird in SMG with the IR now affected by shake, could this be made toggleable?

@Miksel12 Shaking would affect the cursor in real life. :P You can configure just the Y axis (forward/backward) to induce minimal cursor movement. SMG still detects shakes on just Y. You can also reduce the intensity down pretty low for many games. I'd rather not clutter the UI with another option if it's not really needed.

**Comment by Miksel12** [comment]
> @Miksel12 Shaking would affect the cursor in real life. :P You can configure just the Y axis (forward/backward) to induce minimal cursor movement. SMG still detects shakes on just Y. You can also reduce the intensity down pretty low for many games. I'd rather not clutter the UI with another option if it's not really needed.

That is understandable, the auto hide function also alleviates the problems for the most part so it's not a big deal.

**Comment by JMC47** [comment]
Retested and seems good.  People will need to get used to certain motions being more logical though, like shake affecting IR position.

**Pull request merged**